indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -514,6 +518,9 @@ public TaskStatus runTask(TaskToolbox toolbox) throws Exception
               failCnt++;
               log.warn("Failed to run indexSpec: [%s].\nTrying the next indexSpec.", json);
             }
+            Optional.ofNullable(eachSpec.getCompletionReports())
+                    .ifPresent(reports -> completionReports.putAll(
+                        CollectionUtils.mapKeys(reports, key -> getReportkey(eachSpec.getBaseSubtaskSpecName(), key))));
```

Comment:
do we need this getBaseSubtaskSpecName logic? isn't it just the the id of the CompactionTask?

Reply:
`getBaseSubtaskSpecName` already includes the `0` and `1`s. i was trying to use the same indexes here rather than generate them again (bcoz anything throws exception in middle we would deviate from the actual index.)

Reply:
is it okay to just use an indexed for loop ?